### ISSUE TITLE
kdeplot "cut" not as advertised when statsmodels installed; cut=0 leads to wrapped plots

### ISSUE THREAD

**Issue by mdickinson** [mwaskom/seaborn#938]
When performing a univariate KDE plot backed by statsmodels, passing `cut=0` results in the plot being wrapped.

Example plot and code: here's a plot obtained with the statsmodels-backed KDE functionality. The underlying distribution should have a decreasing tail to the right; the jump in the graph (circled in red) is misleading.

![screen shot 2016-06-03 at 13 43 20](https://cloud.githubusercontent.com/assets/662003/15779422/1b546cb0-2994-11e6-8999-f642a612928a.png)

And here's the corresponding plot obtained without statsmodels:

![screen shot 2016-06-03 at 13 42 56](https://cloud.githubusercontent.com/assets/662003/15779356/b5d0570a-2993-11e6-9727-4ccbc7346786.png)

Here's the code that generated the above:

``` python
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats
import seaborn as sns

sample_count = 10**5
x_dist = scipy.stats.lognorm(3.4, loc=0.0, scale=10.0)
y_dist = scipy.stats.lognorm(5.2, loc=0.0, scale=23.0)

x = x_dist.rvs(sample_count).astype(np.float32)
y = y_dist.rvs(sample_count).astype(np.float32)

xy = x * y
xy_exact = x.astype(np.float64) * y.astype(np.float64)
abs_error = abs(xy_exact - xy)
rel_error = abs_error / xy_exact

plt.figure()
sns.kdeplot(rel_error, shade=True, cut=0.0)
plt.show()
```

Diagnosis: when statsmodels is installed, the `cut` parameter passed to `kdeplot` gets [passed straight through](https://github.com/mwaskom/seaborn/blob/69e7f371d27725160d092a528c96cf1fce99b8b4/seaborn/distributions.py#L269) to statsmodels `KDEUnivariate.fit`, but it looks to me as though the latter `cut` parameter has a different meaning from the `kdeplot` one: it's used to add sufficient padding for the FFT operations so that the above wraparound is avoided.

Python and package versions: Python 2.7.11, seaborn 0.7.0, statsmodels 0.6.1, scipy 0.17.0

**Comment by mwaskom** [comment]
Could be a change in statsmodels, pretty sure the manual implementation of "cut" for the scipy backend was originally intended to replicate statsmodels behavior.

**Comment by mwaskom** [comment]
Or maybe I'm thinking of `clip`...

**Comment by mwaskom** [comment]
I have been thinking about removing the statsmodels-based kde fitting anyway; it's not clear that the fft algorithm is all that helpful for the size of datasets that people would be trying to visualize, and there are too many annoying little inconsistencies that pop up.

**Comment by YonelaNuba** [comment]
Hi guys I am also having a problem with my graphs I don't know what's happening please help here is the code of the first graph
plt.figure(figsize = (10, 8))

# KDE plot of loans that were repaid on time
sns.kdeplot(train.loc[train['TARGET'] == 0, 'DAYS_BIRTH'] / 365, label = 'target == 0')

# KDE plot of loans which were not repaid on time
sns.kdeplot(train.loc[train['TARGET'] == 1, 'DAYS_BIRTH'] / 365, label = 'target == 1')

# Labeling of plot
plt.xlabel('Age (years)'); plt.ylabel('Density'); plt.title('Distribution of Ages');
![capture](https://user-images.githubusercontent.com/38220900/44329973-28078100-a466-11e8-8934-2068b71563df.PNG)

**Comment by mwaskom** [comment]
Closed with #2104 with the removal of the statsmodels backend
